api/modules/contacts/cotact.controller.js

```diff
@@ -0,0 +1,169 @@
+const express = require('express');
+const responseBody = require('../../utils/responseBody');
+const logger = require('../../utils/logger');
+const validateSchema = require('../../utils/validateSchema');
+const contactModel = require('./contacts.model');
+const contactSchemas = require('./contactValidationSchema');
+
+const router = express.Router();
+
+/**
+ * retreve all contacts
+ * [TODO] separate logic from the controller
+ */
+router.get('/', async (_req, res) => {
+  try {
+    const contacts = await contactModel.find();
+
+    const resMesg = 'contacts retrived successfuly';
+    const resBody = responseBody(resMesg, contacts);
+
+    return res.json(resBody);
+  } catch (err) {
+    logger.error(`error [GET] @contacts controller: -> ${err}`);
+    const resBody = responseBody('Server Error');
```

Comment:
I think in this situation (retreving data) most errors are server errors, the only situation (as I think), when there is no data in the db, in this situation I send empty array instead of error message, and I do not know the perfect or suitable.

But if you ask about all controllers, I usually put the server error first, then test and debug to get other cases, but here I do not use real database